### ISSUE TITLE
Proposal: If a contributor unstakes an open pooled Service Node it will immediately jump into the service node awaiting contribution list.

### ISSUE THREAD

**Issue by Sonofotis** [oxen-io/oxen-core#817]
# TLDR
Keep pooled Service Nodes registered by allowing others to stake to the Service Node the amount that was requested to be unlocked.

## Problem
When a contributor runs `request_stake_unlock` on an open pool Service Node it is deregistered in 15 days and the operator must reregister the Service Node and await new contributors.

This means the Service Node will be removed from it's swarm and other duties until the operator reregistered the SN and new contributors fill up the remaining contribution.

## Proposal
Once a contributor requests a stake to be unlocked we can keep the Service Node running by moving the SN to the "Awaiting Contribution list" for the 15 days unlock period. If someone stakes to the SN within that 15 days the SN does not deregister and continues as normal. 

### Additional points:
- The contributor who unlocks will still receive the rewards for the 15 days. 
- New contributors will need to wait till the 15 days are up before they receive rewards.
- Lokiblocks will display if a pooled SN is "New" or "Old" and will only show "Old" SN's when they are at the last 2 days of their unlock period.
- If no one contributes to the open slot within the 15 days the SN will be deregistered as normal.
- The staking requirement will be changed to the LSR for the BH in which the SN would have deregistered.
  - Example: Unlock stake @ BH 200,000. The new LSR is worked out for BH 200,000 + 15(days)*720(blocks per day).
- The operator and old contributors will have a larger share of the rewards when the LSR changes. 
  - Example: LSR = 20k, Operator and 3 contributors have 5000 staked each.
  - The new LSR is now 18k because a contributor unstaked, which means there is only 3k open for contribution as the operator has 5k locked and the two other old contributors have 5k locked aswell.

**Comment by CryptoFirefly** [comment]
What additionally Needs to be considered is how the Operator can override during the unlock phase, for e.g. if the Op wants to change his/her stake % or make amendments to the Op fee. How would those work if someone else already contributed to fill the newly vacant space. 

Also currently as the Stakes are dropping this may not be as attractive for new contributors, as they would prefer to stake into Nodes with less total stake, so their Loki amount gets a higher total % in a node, increasing the rewards for them.
Hence this definitely is something we could have once the Stake per node is fixed in a year's time.

**Comment by Sonofotis** [comment]
@CryptoFirefly We would assume the operator does not want to change the operator fee until they want to unlock their stake. At this point the node will deregister as normal and the operator can register a SN with a different Op fee and % stake.

"as they would prefer to stake into Nodes with less total stake," 
This problem is an issue for new SN's aswell. If an operator commits 25% stake and a contributor does 72% there would still need to be a 3rd operator to complete the remaining 3%.

**Comment by Doy-lee** [comment]
The overall scenario doesn't really make sense to go and queue up in a pooled node locking your funds when you could leave that Loki readily available to jump onto another pooled node if it shows up before to get staking rewards sooner.

Especially the awkwardness around mandatorily having to wait the 15 days for the other contributors funds the unlock. To make it less awkward would be allowing the new contributor to unlock their contribution at will (which is abusable by spamming the blockchain with this special TX, currently unlocking funds is not abusable due to the 15 day unlock period) or that if the node deregisters their funds don't get locked. Just returned.

But if thats the case then why queue up in a node when you'd be better off waiting until the last day to stake in or stake into a new pooled node. I think that being able to stake in on the last day so there's minimal lock time and preserve the Service Node on the network might have some merit in improving the UX here.

**Comment by darcys22** [comment]
Discussion about feature of this nature occurring here:
https://github.com/oxen-io/oxen-improvement-proposals/issues/27
